auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -704,4 +750,101 @@ private static void buildParameterizedTypeArguments(CodeBlock.Builder block, Typ
       block.add("$T.class", typeArg);
     }
   }
+
+  private static Optional<ExecutableElement> findDefaultBuilder(TypeElement autoValueClass, TypeElement builderClass) {
+    return ElementFilter.methodsIn(autoValueClass.getEnclosedElements())
+        .stream()
+        .filter(e -> MoreElements.isAnnotationPresent(e, GsonDefaultBuilder.class))
+        .filter(MoreElements.hasModifiers(Modifier.STATIC)::apply)
+        .filter(e -> MoreTypes.equivalence().equivalent(e.getReturnType(), builderClass.asType()))
+        .filter(e -> e.getParameters().isEmpty())
+        .findFirst();
+  }
+
+  /**
+   * Searches the builder for an method has no parameters, returns the autovalue class, is not
+   * private, and is not static. If there is a valid method annotated with {@link GsonBuild}, return
+   * that method. Otherwise, return the method named "build" if present.
+   */
+  private static Optional<ExecutableElement> findBuildMethod(
```

Comment:
This should follow the same method that AutoValue uses to identify build methods in a builder.  That is, a non-private, non-static method with no args that returns the annotated type.  The method need not be named `build` (`create` is another common option).  If there is more than one (both `create` and `build` and `createWithDefaults` and `createByIncrementingCount`, etc) then you can use the annotation to decide which one.